src/test/ui/nll/issue-21232-partial-init-and-erroneous-use.rs

```diff
@@ -0,0 +1,60 @@
+// This test enumerates various cases of interest where a ADT or tuple is
+// partially initialized and then used in some way that is wrong *even*
+// after rust-lang/rust#54987 is implemented.
+//
+// See rust-lang/rust#21232, rust-lang/rust#54986, and rust-lang/rust#54987.
+//
+// See issue-21232-partial-init-and-use.rs for cases of tests that are
+// meant to compile and run successfully once rust-lang/rust#54987 is
+// implemented.
+
+#![feature(nll)]
+
+struct D {
+    x: u32,
+    s: S,
+}
+
+struct S {
+    y: u32,
+    z: u32,
+}
+
+
+impl Drop for D {
+    fn drop(&mut self) { }
+}
+
+fn cannot_partially_init_adt_with_drop() {
+    let d: D;
+    d.x = 10;
```

Comment:
Please add `//~ ERROR` annotations where appropriate (why aren't these required, anyway? probably my fault)

Reply:
Hmm you’re right, why aren’t they required here... (is it one of those goofy things where if none of a particular diagnostic class are present then none are required?)

Reply:
(okay added them here and also to the way more meaty ones in borrowck for issue #54499 .)